src/librustdoc/html/templates/page.html

```diff
@@ -13,13 +13,23 @@
           href="{{static_root_path | safe}}rustdoc{{page.resource_suffix}}.css" {# -#}
           id="mainThemeStyle"> {#- -#}
     {{- style_files | safe -}}
-    <script id="default-settings" {# -#}
+    <script defer id="default-settings" {# -#}
       {% for k, v in layout.default_settings %}
         data-{{k}}="{{v}}"
       {%- endfor -%}
     ></script> {#- -#}
-    <script src="{{static_root_path | safe}}storage{{page.resource_suffix}}.js"></script> {#- -#}
-    <script src="{{page.root_path | safe}}crates{{page.resource_suffix}}.js"></script> {#- -#}
+    <script defer src="{{static_root_path | safe}}storage{{page.resource_suffix}}.js"></script> {#- -#}
+    <script defer src="{{page.root_path | safe}}crates{{page.resource_suffix}}.js"></script> {#- -#}
```

Comment:
Also, storage and crates JS should not get the defer attribute. They are both on the critical path for rendering the page, so we actually want them to block page render, rather than cause a re-style later.

As an example of why this is important, try changing to a dark theme, and then reloading the page. If storage has the defer attribute, you'll see the page first load with light styles and then with dark (assuming your system default theme is light).

Reply:
@jsha 
Thank you so much for your extremely clear review! I've made changes as you show :)